packages/wrangler/pages/functions/filepath-routing.test.ts

```diff
@@ -83,34 +83,133 @@ describe("filepath-routing", () => {
       `
       );
 
+      mkdirSync("todos");
+
+      writeFileSync(
+        "todos/[id].ts",
+        `
+      export function onRequestPost() {}
+      export function onRequestDelete() {}
+      `
+      );
+
+      mkdirSync("authors");
+      mkdirSync("authors/[authorId]");
+      mkdirSync("authors/[authorId]/todos");
+
+      writeFileSync(
+        "authors/[authorId]/todos/[todoId].ts",
+        `
+      export function onRequestPost() {}
+      `
+      );
+
+      mkdirSync("books");
+
+      writeFileSync(
+        "books/[[title]].ts",
+        `
+      export function onRequestPost() {}
+      `
+      );
+
+      mkdirSync("cats");
+      mkdirSync("cats/[[breed]]");
+
+      writeFileSync(
+        "cats/[[breed]]/blah.ts",
+        `
+      export function onRequestPost() {}
+      `
+      );
+
+      writeFileSync(
+        "cats/[[breed]]/[[name]].ts",
+        `
+      export function onRequestPost() {}
+      `
+      );
+
       const entries = await generateConfigFromFileTree({
         baseDir: ".",
         baseURL: "/base" as UrlPath,
       });
-      expect(entries).toEqual({
-        routes: [
-          {
-            method: "DELETE",
-            module: ["bar.ts:onRequestDelete"],
-            routePath: "/base/bar",
-          },
-          {
-            method: "PUT",
-            module: ["bar.ts:onRequestPut"],
-            routePath: "/base/bar",
-          },
-          {
-            method: "GET",
-            module: ["foo.ts:onRequestGet"],
-            routePath: "/base/foo",
-          },
-          {
-            method: "POST",
-            module: ["foo.ts:onRequestPost"],
-            routePath: "/base/foo",
-          },
-        ],
-      });
+      expect(entries).toMatchInlineSnapshot(`
+        Object {
+          "routes": Array [
+            Object {
+              "method": "POST",
+              "module": Array [
+                "authors/[authorId]/todos/[todoId].ts:onRequestPost",
+              ],
+              "routePath": "/base/authors/:authorId/todos/:todoId",
+            },
+            Object {
+              "method": "POST",
+              "module": Array [
+                "cats/[[breed]]/blah.ts:onRequestPost",
+              ],
+              "routePath": "/base/cats/:breed*/blah",
+            },
+            Object {
+              "method": "POST",
+              "module": Array [
+                "cats/[[breed]]/[[name]].ts:onRequestPost",
+              ],
+              "routePath": "/base/cats/:breed*/:name*",
```

Comment:
How would this case work in practice? What would map to the two different catch alls?

Reply:
This won't actually work at runtime. Neither will something with a fixed separator between like `/[[foo]]/fixed/[[bar]].ts`.

Currently these [[catch-all]] parts need to be only the terminal. But maybe we change the behavior in the future to be non-greedy? Either way, we thought it made sense to at least fix the parsing while we were here.

Reply:
Would it not be a better DX if `pages` was able to give an error (or warning) if it finds these cases?

Reply:
Honestly, yeah... But I reckon we could do that in a later PR? This doesn't _add_ that behaviour, just fixes the parsing.